spring-integration-core/src/test/java/org/springframework/integration/dsl/routers/RouterTests.java

```diff
@@ -612,6 +613,21 @@ public void testNestedScatterGatherSuccess() {
 
 	}
 
+	@Timeout(11000)
+	@Test
+	public void testNestedScatterGatherSequenceTest() {
+		PollableChannel replyChannel = new QueueChannel();
+		this.scatterGatherInSubFlowChannel.send(
+				MessageBuilder.withPayload("sequencetest")
+						.setReplyChannel(replyChannel)
+						.build());
+
+		Message<?> receive = replyChannel.receive(10000);
+		assertThat(receive).isNotNull();
+		assertThat(receive.getPayload()).isEqualTo("sequencetest");
```

Comment:
Excuse me, and how with this test we can be sure that sequence is popped properly. Doesn't look like other test have some difference with this one, but they don't fails...

Reply:
Added additional condition checks to ensure Sequence is properly popped.